### DISCUSSION TITLE
macaroon

### DISCUSSION THREAD

**Discussion by amicwall** [Ride-The-Lightning/c-lightning-REST#123] [Q&A]
in browser: http://ip:port/v1/getinfo
result = {"message":"Authentication Failed!","error":"Bad Macaroon!"}

**Comment by ShahanaFarooqui** [comment]
Please try to post a complete question with related details.

**Comment by ShahanaFarooqui** [comment]
From the first glance, the first line has 2 problems:

1) The url should start with ‘https’ not with ‘http’.
2) This error is an expected behavior in this scenario because macaroon is missing from the header.

**Reply by amicwall** [reply]
my macaroon got generated in: /home/coin/rtl/Rest/c-lightning-REST-0.7.2/certs
but apparently it can't find it there. Since I can't find the log file, I can't see whats going on

**Comment by mcroni** [comment]
i also get this same error even with using the hex value of the macaroon file.

**Reply by ShahanaFarooqui** [reply]
We need more information to debug the issue or suggest anything further. Like, what tool did you use to call the API (curl, postman, browser), what was the url, what is the location of the macaroon, what are your Core lightning and c-lightning-REST versions etc.
